src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,138 @@
+            Factory<TSerializer> serializerFactory)
+            where TSerializer : class, ISerializer
+        {
+            return middlewares.AddSerializer(
+                serializerFactory,
+                _ => new DefaultMessageTypeResolver());
+        }
+
+        /// <summary>
+        /// Register a middleware to serialize the message to a fixed type
+        /// </summary>
+        /// <param name="middlewares">The middleware configuration builder</param>
+        /// <param name="serializerFactory">A factory to create a <see cref="ISerializer"/></param>
+        /// <typeparam name="TMessage">The message type</typeparam>
+        /// <typeparam name="TSerializer">A class that implements <see cref="ISerializer"/></typeparam>
+        /// <returns></returns>
+        public static IProducerMiddlewareConfigurationBuilder AddSingleTypeSerializer<TMessage, TSerializer>(
+            this IProducerMiddlewareConfigurationBuilder middlewares,
+            Factory<TSerializer> serializerFactory)
+            where TSerializer : class, ISerializer
+        {
+            return middlewares.AddSerializer(
+                serializerFactory,
+                _ => new SingleMessageTypeResolver(typeof(TMessage)));
+        }
+
+        /// <summary>
+        /// Register a middleware to serialize the message to a fixed type
+        /// </summary>
+        /// <param name="middlewares">The middleware configuration builder</param>
+        /// <typeparam name="TMessage">The message type</typeparam>
+        /// <typeparam name="TSerializer">A class that implements <see cref="ISerializer"/></typeparam>
+        /// <returns></returns>
+        public static IProducerMiddlewareConfigurationBuilder AddSingleTypeSerializer<TMessage, TSerializer>(
+            this IProducerMiddlewareConfigurationBuilder middlewares)
+            where TSerializer : class, ISerializer
+        {
+            return middlewares.AddSingleTypeSerializer<TSerializer>(typeof(TMessage));
+        }
+
+        /// <summary>
+        /// Register a middleware to serialize the message to a fixed type
+        /// </summary>
+        /// <param name="middlewares">The middleware configuration builder</param>
+        /// <param name="messageType">The message type</param>
+        /// <typeparam name="TSerializer">A class that implements <see cref="ISerializer"/></typeparam>
+        /// <returns></returns>
+        public static IProducerMiddlewareConfigurationBuilder AddSingleTypeSerializer<TSerializer>(
+            this IProducerMiddlewareConfigurationBuilder middlewares,
+            Type messageType)
+            where TSerializer : class, ISerializer
+        {
+            middlewares.DependencyConfigurator.AddTransient<TSerializer>();
+
+            return middlewares.AddSerializer(
+                resolver => resolver.Resolve<TSerializer>(),
+                _ => new SingleMessageTypeResolver(messageType));
+        }
+    }
+}
```

Comment:
Missing an overload where we can pass the Type instance and a serializer Factory in Consumer and Producer builder

Reply:
Sure, added it